### PR TITLE
feat: generate Storagefield for form builder

### PR DESCRIPTION
**Pull request by bombguy**
## Problem
To support generate StorageField configured by form builder

## Additional Notes

- snapshot of the generate code
- example data schema & form definition
- workaround on the stringified array field

### Manual tests
Manually tested the generated code by modifying the data schema & form definition.

### Automated tests
- [X] Unit tests added/updated
- [ ] E2E tests added/updated - Deferred integ testing.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/codegen-ui-react/package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
   },
   "dependencies": {
     "@aws-amplify/codegen-ui": "2.11.1",
+    "@aws-amplify/ui-react-storage": "^0.0.0-storagemanager-37cac7b-20230403170123",
```
Comment (bombguy) [inline packages/codegen-ui-react/package.json:34]: Will revisit this when we the package is published

File: packages/codegen-ui-react/lib/utils/forms/storage-field-component.ts [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+          factory.createJsxAttribute(
+            factory.createIdentifier(key),
+            factory.createJsxExpression(undefined, propertyToExpression(componentMetadata, value)),
+          ),
+        );
+      }
+    }
+    if (isStorageManagerKey(key)) {
+      const storageManagerValue =
+        key === 'acceptedFileTypes' ? { ...value, value: (value as any).value.split(',') } : value;
```
Comment (bombguy) [inline packages/codegen-ui-react/lib/utils/forms/storage-field-component.ts:170]: When the form definition gets passed to the codegen-ui-react, the values get stringified on `mapFormElementProps`. This is a sensitive code that will require more in-depth refactoring to change.

This limits the blast radius to this particular case. We should revisit this later to refactor mapFormElementProps.

File: packages/codegen-ui-react/lib/react-component-render-helper.ts [hunk 3]
```diff
@@ -193,12 +193,14 @@ export function buildFixedLiteralExpression(
...
   switch (typeof value) {
     case 'number':
-      return factory.createNumericLiteral(value, undefined);
+      return factory.createNumericLiteral(value as number, undefined);
     case 'boolean':
       return value ? factory.createTrue() : factory.createFalse();
     case 'string':
-      return fixedPropertyWithTypeToLiteral(value, type);
     case 'object':
+      if (type !== 'object') {
```
Comment (bombguy) [inline packages/codegen-ui-react/lib/react-component-render-helper.ts:201]: related to the stringified array issue on comment below. Workaround to reduce blast radius.

File: packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap [hunk 4]
```diff
@@ -24794,6 +24794,423 @@ export default function MyMemberForm(props: MyMemberFormProps): React.ReactEleme
...
+        hasError={errors.imgKeys?.hasError}
+        label={\\"Images\\"}
+        isRequired={false}
+        isReadOnly={false}
+      >
+        <StorageManager
+          ref={imgKeysRef}
+          defaultFiles={imgKeys.map((s3Key) => ({ s3Key }))}
+          onFileSuccess={(files) => {
+            let value = files.map(({ s3Key }) => s3Key);
```
Comment (hein-j) [inline packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap:25158]: What happens when the field is not an array? Or are we just going to allow only for string[]? Not string?

Comment (bombguy) [reply packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap:25158]: I will add the unit test & impl for the non-array field as well on next commit.

**Review by hein-j — APPROVED** [review]

**Review by rtpascual — DISMISSED** [review]

**Review by rtpascual — DISMISSED** [review]

**Review by hein-j — DISMISSED** [review]

**Review by hein-j — DISMISSED** [review]

**Review by hein-j — APPROVED** [review]

**Review by rtpascual — APPROVED** [review]

**Pull request merged**